Source/Core/UICommon/UICommon.cpp

```diff
@@ -225,4 +230,24 @@ void SaveWiimoteSources()
   inifile.Save(ini_filename);
 }
 
+bool TriggerSTMPowerEvent(bool use_debugger)
+{
+  const auto ios = IOS::HLE::GetIOS();
+  if (!ios)
+    return false;
+
+  const auto stm = ios->GetDeviceByName("/dev/stm/eventhook");
+  if (!stm || !std::static_pointer_cast<IOS::HLE::Device::STMEventHook>(stm)->HasHookInstalled())
+    return false;
+
+  Core::DisplayMessage("Shutting down", 30000);
+  // Unpause because gracefully shutting down needs the game to actually request a shutdown.
+  // Do not unpause in debug mode to allow debugging until the complete shutdown.
+  if (Core::GetState() == Core::State::Paused && !use_debugger)
```

Comment:
It would have to call ``ProcessorInterface::PowerButton_Tap()`` after that, which makes it kinda clumsy.